packages/mobile/src/web3/saga.ts

```diff
@@ -31,47 +33,65 @@ import {
 } from 'src/web3/actions'
 import { web3 } from 'src/web3/contracts'
 import { refreshGasPrice } from 'src/web3/gas'
+import { web3ReadySelector } from 'src/web3/reducer'
 import { currentAccountSelector } from 'src/web3/selectors'
 import { Block } from 'web3/eth/types'
 
 const ETH_PRIVATE_KEY_LENGTH = 64
 const MNEMONIC_BIT_LENGTH = (ETH_PRIVATE_KEY_LENGTH * 8) / 2
 
 const TAG = 'web3/saga'
-// The timeout for web3 to complete syncing
-const CHECK_WEB3_SYNC_PROGRESS_TIMEOUT = 60000
 // The timeout for web3 to complete syncing and the latestBlock to be > 0
 const CHECK_SYNC_PROGRESS_TIMEOUT = 60000
 const BLOCK_CHAIN_CORRUPTION_ERROR = "Error: CONNECTION ERROR: Couldn't connect to node on IPC."
 
 let AssignAccountLock = false
 
+export function* waitForWeb3Ready(): any {
+  const connected = yield select(web3ReadySelector)
+  if (connected) {
+    return
+  }
+
+  while (true) {
+    const action = yield take(Actions.SET_IS_READY)
+    if (action.payload.isReady) {
+      return
+    }
+  }
+}
+
 // checks if web3 claims it is currently syncing or not
-function* checkWeb3SyncProgressClaim() {
-  Logger.debug(TAG, 'Checking sync progress claim')
+export function* checkWeb3SyncProgressClaim() {
+  // yield call(waitForGethConnectivity)
   while (true) {
+    yield delay(1000) // Doesn't need to update that fast
```

Comment:
what's this for?

Reply:
Because it's inside of a while loop that's triggered ass soon as the app starts. Without delay the app would be looping to fast and spending too much resources pooling Web3 status. Maybe it'd be better if we trigger this check from `waitForNetwork` to avoid pooling Web3.

My solution has a worse case were Web3 gets disconnected just before checking for `isReady` and before the next `web3.eth.isSyncing()` call.